File: SpringBootJpa/src/main/java/com/example/simple/domain/TimeData.java
package com.example.simple.domain;

import lombok.Data;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;

@Entity
@Data
@RequiredArgsConstructor
public class TimeData implements Serializable {

    @Id @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long idx;

    @NonNull
//    @Temporal(TemporalType.DATE)
    private LocalDateTime date;
    
    public TimeData() {}
}

File: SpringBootJpa/src/main/kotlin/com/example/kotlin/simple/domain/TimeData.kt
package com.example.kotlin.simple.domain

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import java.time.LocalDateTime

@Entity
data class TimeData(
        @Id @GeneratedValue val idx: Long? = null,
        val date: LocalDateTime)
